### PR TITLE
Make hardware triggers match latest spec.

### PR DESCRIPTION
**Pull request by timsifive**

### CODE REVIEW

**Comment by aswaterman** [comment]
updated per discussions today

File: encoding.h [hunk 1]
```diff
@@ -68,6 +68,10 @@
 #define MCONTROL_STORE      (1<<1)
 #define MCONTROL_LOAD       (1<<0)
 
+#define MCONTROL_MASKMAX(xlen) (0x3f<<((xlen)-11))
```
Comment (timsifive) [inline encoding.h]: I wasn't sure what the convention for those was. This is a good solution. I'll change the spike code to use them.

**Comment by timsifive** [comment]
Andrew's changes look good.

File: encoding.h (hunk 1 above)
Comment (aswaterman) [inline encoding.h]: yeah, not sure there /was/ a convention...

**Comment by aswaterman** [comment]
does this look right?

**Comment by timsifive** [comment]
I think this captures our e-mail discussion and my spec updates.

**Comment by aswaterman** [comment]
merged via 1a25601e9bbc2a0ab49203ef7a0b6d5a44c4b658

**Pull request closed without merging**